platform/api/esx/api.go

```diff
@@ -0,0 +1,742 @@
+
+	task, err := baseVM.Clone(a.ctx, folder, name, *cloneSpec)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't clone base VM: %v", err)
+	}
+
+	err = task.Wait(a.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("clone base VM operation failed: %v", err)
+	}
+
+	vm, err := defaults.finder.VirtualMachine(a.ctx, name)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't find cloned VM: %v", err)
+	}
+
+	err = a.addSerialPort(vm)
+	if err != nil {
+		return nil, fmt.Errorf("adding serial port: %v", err)
+	}
+
+	err = a.updateOVFEnv(vm, ignitionData)
+	if err != nil {
+		return nil, fmt.Errorf("setting guestinfo settings: %v", err)
+	}
+
+	err = a.startVM(vm)
+	if err != nil {
+		return nil, fmt.Errorf("starting vm: %v", err)
+	}
+
+	mach, err := a.getMachine(vm)
+	if err != nil {
+		return nil, fmt.Errorf("getting machine info: %v", err)
+	}
+
+	return mach, nil
+}
+
+func (a *API) CreateBaseDevice(name, ovaPath string) error {
+	defaults, err := a.getServerDefaults()
+	if err != nil {
+		return fmt.Errorf("getting ESX defaults: %v", err)
+	}
+
+	if ovaPath == "" {
+		ovaPath = filepath.Join(sdk.BuildRoot(), "images", "amd64-usr", "latest", "coreos_production_vmware_ova.ova")
+	}
+
+	archive, cisr, err := a.buildCISR(name, ovaPath, defaults.finder, defaults.network, defaults.resourcePool, defaults.datastore)
+	if err != nil {
+		return fmt.Errorf("building CreateImportSpecRequest: %v", err)
+	}
+
+	folders, err := defaults.datacenter.Folders(a.ctx)
+	if err != nil {
+		return fmt.Errorf("getting datacenter folders: %v", err)
+	}
+	folder := folders.VmFolder
+
+	entity, err := a.uploadToResourcePool(archive, defaults.resourcePool, cisr, folder)
+	if err != nil {
+		return fmt.Errorf("uploading disks to ResourcePool: %v", err)
+	}
+	_ = object.NewVirtualMachine(a.client.Client, *entity)
+
+	return nil
+}
+
+func (a *API) TerminateDevice(vmName string) error {
+	defaults, err := a.getServerDefaults()
+	if err != nil {
+		return fmt.Errorf("couldn't get server defaults: %v", err)
+	}
+
+	vm, err := defaults.finder.VirtualMachine(a.ctx, vmName)
+	if err != nil {
+		return fmt.Errorf("couldn't find VM: %v", err)
+	}
+
+	return a.deleteDevice(vm)
+}
+
+func (a *API) deleteDevice(vm *object.VirtualMachine) error {
+	task, err := vm.PowerOff(a.ctx)
+	if err != nil {
+		return fmt.Errorf("powering off vm: %v", err)
+	}
+
+	// We don't check for errors on this task because it will throw an error
+	// if the VM is already in a powered off state
+	_ = task.Wait(a.ctx)
+
+	task, err = vm.Destroy(a.ctx)
+	if err != nil {
+		return fmt.Errorf("destroying vm: %v", vm)
+	}
+
+	return task.Wait(a.ctx)
+}
+
+func (a *API) buildCISR(name string, flagOVAPath string, finder *find.Finder, defaultNetwork object.NetworkReference, resourcePool *object.ResourcePool, datastore *object.Datastore) (*Archive, *types.OvfCreateImportSpecResult, error) {
+	var nets []types.OvfNetworkMapping
+	nets = append(nets, types.OvfNetworkMapping{
+		Name:    "test",
```

Comment:
Is this the intended name?  If you need something specific to this codebase, consider `mantle`.